File: easy/find_the_highest_score.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

func max(a, b int) int {
	if b > a {
		return b
	}
	return a
}

func main() {
	var a int
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()
	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		var (
			r []int
			u []string
		)
		s := strings.Split(scanner.Text(), "|")
		for ix, i := range s {
			t := strings.Fields(strings.TrimSpace(i))
			for jx, j := range t {
				fmt.Sscanf(j, "%d", &a)
				if ix == 0 {
					r = append(r, a)
				} else {
					r[jx] = max(r[jx], a)
				}
			}
		}
		for _, i := range r {
			u = append(u, fmt.Sprint(i))
		}
		fmt.Println(strings.Join(u, " "))
	}
}

File: easy/find_the_highest_score.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split(" | ").map { |x| x.split.map(&:to_i) }
  for i in 1...s.length
    s[i].each_with_index do |x, j|
      s[0][j] = x if s[0][j] < x
    end
  end
  puts s[0].join(" ")
end
